packages/data-stores/src/products-list/resolvers.ts

```diff
@@ -0,0 +1,33 @@
+import { wpcomRequest } from '../wpcom-request-controls';
+import { requestProductsList, receiveProductsList, receiveProductsListFailure } from './actions';
+import { ensureNumericCost } from './assembler';
+import { ProductsList, ProductsListFailure } from './types';
+
+/**
+ * Requests the list of all products from the WPCOM API.
+ */
+export function* getProductsList() {
+	yield requestProductsList();
+
+	try {
+		const productsList: ProductsList = yield wpcomRequest( {
+			path: '/products',
+			apiVersion: '1.1',
+		} );
+
+		// Since the request succeeded, productsList should be guaranteed non-null;
+		// thus, we don't have any safety checks before this line.
+
+		// Create a completely new products list to avoid mutating the original
+		const sanitizedProductsList = Object.fromEntries(
+			Object.entries( productsList ).map( ( [ slug, product ] ) => [
+				slug,
+				ensureNumericCost( product ),
+			] )
+		);
```

Comment:
Optional: since mutation is no longer a concern, I'd simplify this to:

```suggestion
		for(const product of Object.values(productList)) {
			product.cost = Number(product.cost)
		}
```

And naturally delete `assembler.ts`.